### ISSUE TITLE
Классификатор банков РБ

### ISSUE THREAD

**Issue by chertby** [minsk-hackerspace/api.nbrb.by#12]
В большинстве учетных систем (1С, SAP и так далее) используется классификатор банков. На данный момент его либо вводят руками, либо загружали из файлы excel, который публиковали на сайте www.nbrb.by. После перехода на IBAN, файл перестали публиковать, зато появилось API http://www.nbrb.by/apihelp/bic. Можно было бы использовать данный API, но в нем не хватает данных.

Я предлагаю сделать API для получения классификатора банков РБ.

nbrb.openapi.by/banks

```
{
"banks": {
    "bic": "string",
    "active": true,
    "name": "string",
    "city": "string",
    "address": "string",
    "phone": "string"
  }
}
```

  - [x] Сделать копию базы
  - [x] Сделать API
  - [x] Сделать пример

**Comment by bamunra** [comment]
Проблема в том, что сам API мы сделать не можем. Мы не являемся хозяевами классификатора. Да, ты можешь скопировать каким то образом базу к себе и написать для неё API и отдать. Но за актуальность данных в базе никто не поручится. Так как API банковское, то и писать их должны банки.

**Comment by chertby** [comment]
Согласен. Но думаю, пока можно скопировать данные, что мы и сделали) [nbrb.openapi.by/banks.csv.](https://github.com/openapiby/nbrb.openapi.by/blob/master/banks.csv) Далее сделать API и пример как его можно использовать. А там 2 вариант либо НБРБ проникнется либо поддерживать в актуальном состоянии, благо он редко меняется.

**Comment by chertby** [comment]
Добавил первую версию описания API [nbrb.openapi.by/spec/swagger.yaml.](https://github.com/openapiby/nbrb.openapi.by/blob/master/spec/swagger.yaml)

**Comment by R0bur** [comment]
Посмотрите ещё здесь: [http://banks.of.by/bic.php](http://banks.of.by/bic.php)
В отличие от классификатора НБРБ, здесь приводится расширенная информация по филиалам Беларусбанка, которые имеют индивидуальные БИК. Правда, филиалы эти потихоньку выводятся из справочника...

**Comment by abitrolly** [comment]
@R0bur а это случайно не копия https://www.nbrb.by/engl/payment/Bic?page=0&Mode=-1&PageSize=1000&Order=ASC&SortField=typ ? Там тоже 75 позиций.

Интересно, что http://www.nbrb.by/api/bic выдаёт 112 позиций. Попробую сравнить.

**Comment by R0bur** [comment]
@abitrolly:
> а это случайно не копия

Не копия, хотя за основу, конечно, взят классификатор НБРБ.

> Интересно, что http://www.nbrb.by/api/bic выдаёт 112 позиций.

Этот API выдаёт не только актуальное состояние справочника, но и некоторую историю. В нём есть несколько записей с одинаковыми БИК. Возможно, так и было задумано, а может, это ошибка реализации.

**Comment by chertby** [comment]
@abitrolly как насчет того что бы сделать web api?
Я могу сделать на .Net core 3.0 добавлю swagger. Для начала сделаю InMemory.

**Comment by abitrolly** [comment]
Да, было бы супер. Я уже забыл что тут и где лежит, поэтому смог только README поправить в #13 .Net core я не знаю, но это и хорошо - буду меньше отвлекаться. :D

**Comment by chertby** [comment]
@abitrolly можешь посмотреть #15

**Comment by abitrolly** [comment]
@chertby посмотрел. Ничыво не панятна в этом дотнеце - многа текста. Мержи, потом перепишем на https://www.starlette.io/schemas/ если чё или время будет. )

**Comment by chertby** [comment]
@abitrolly docker можно на netlify запустить или свой сервер надо?

**Comment by abitrolly** [comment]
@chertby Netlify только статические файлы отдаёт. `docker` это обёртка над линукс контейнерами, поэтому везде, где есть линукс будет работать. Google, Amazon и т.п.
